lib/spack/spack/util/web.py

```diff
@@ -592,10 +584,7 @@ def find_versions_of_archive(
         #   .sha256
         #   .sig
         # However, SourceForge downloads still need to end in '/download'.
-        url_regex += r'(\/download)?'
-        # PyPI adds #sha256=... to the end of the URL
-        url_regex += '(#sha256=.*)?'
-        url_regex += '$'
```

Comment:
I can't really comment on the specifics of these few lines, since I'm not as familiar with their purpose.

But, to me, this appears to be another unrelated change that is being proposed for no obvious reason.  If there's a rationale for changes like these, I'd be happy to consider them.